src/apiserver/api_service_impl.cc

```diff
@@ -0,0 +1,317 @@
+            row->AppendInt32(v.GetInt());
+            break;
+        }
+        case hybridse::sdk::kTypeInt64: {
+            row->AppendInt64(v.GetInt64());
+            break;
+        }
+        case hybridse::sdk::kTypeFloat: {
+            row->AppendFloat(boost::lexical_cast<float>(v.GetDouble()));
+            break;
+        }
+        case hybridse::sdk::kTypeDouble: {
+            row->AppendDouble(v.GetDouble());
+            break;
+        }
+        case hybridse::sdk::kTypeString: {
+            row->Init(boost::lexical_cast<int32_t>(v.GetStringLength()));
+            row->AppendString(v.GetString());
+            break;
+        }
+        case hybridse::sdk::kTypeDate: {
+            std::vector<std::string> parts;
+            ::fedb::base::SplitString(v.GetString(), "-", parts);
+            if (parts.size() != 3) {
+                return false;
+            }
+            auto year = boost::lexical_cast<int32_t>(parts[0]);
+            auto mon = boost::lexical_cast<int32_t>(parts[1]);
+            auto day = boost::lexical_cast<int32_t>(parts[2]);
+            row->AppendDate(year, mon, day);
+            break;
+        }
+        case hybridse::sdk::kTypeTimestamp: {
+            row->AppendTimestamp(v.GetInt64());
+            break;
+        }
+        default:
+            return false;
+    }
+    return true;
+}
+
+void APIServiceImpl::RegisterPut() {
+    provider_.put("/db/:db_name/table/:table_name", [this](const InterfaceProvider::Params& param,
+                                                           const butil::IOBuf& req_body, JsonWriter& writer) {
+        auto err = GeneralError();
+        auto db_it = param.find("db_name");
+        auto table_it = param.find("table_name");
+        if (db_it == param.end() || table_it == param.end()) {
+            writer& err.Set("Invalid path");
+            return;
+        }
+        auto db = db_it->second;
+        auto table = table_it->second;
+
+        // json2doc, then generate an insert sql
+        Document document;
+        if (document.Parse(req_body.to_string().c_str()).HasParseError()) {
+            writer& err.Set("Json parse failed");
+            return;
+        }
+
+        const auto& value = document["value"];
+        // value should be array, and multi put is not supported now
+        if (!value.IsArray() || value.Empty() || value.Size() > 1) {
+            writer& err.Set("Invalid value in body");
+            return;
+        }
+        const auto& arr = value[0];
+        StringBuffer buffer;
+        Writer<StringBuffer> sql_writer(buffer);
+        arr.Accept(sql_writer);
+        std::string line(buffer.GetString());
+        std::string insert_sql = "insert into " + table + " values(" + line.substr(1, line.length() - 2) + ");";
+
+        hybridse::sdk::Status status;
+        auto ok = sql_router_->ExecuteInsert(db, insert_sql, &status);
+        if (ok) {
+            PutResp resp;
+            writer& resp;
```

Comment:
add a layer, e.g. writer >> obj, reader << obj, still use operator& inside. 
More easier for caller.